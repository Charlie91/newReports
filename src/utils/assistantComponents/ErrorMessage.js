import React from 'react';

const ErrorMessage = (props) => (
    <div className="message text-white bg-danger text-center card">
        <div className="card-body card-block">
            {props.text}
        </div>
    </div>
);
export default ErrorMessage;