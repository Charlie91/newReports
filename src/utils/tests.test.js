import React from 'react';
import fetch from 'node-fetch';
import renderer from 'react-test-renderer';
import ErrorMessage from './assistantComponents/ErrorMessage';

function digitCount(number) {
    return number.toString().length;
};

it('renders correctly', () => {
    const tree = renderer
        .create(<ErrorMessage text="Текст ошибки"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('count digit number', () => {
    expect(digitCount(123)).toBe(3);
});


test('logged in succesfully', () => {
    let user = {        //данные тестового пользователя
        login: 'Charlie91',
        pwd: '123456'
    };
    let options = {
        method: 'POST',
        credentials: 'include',
        headers: {
            "Content-type": "application/json;charset=UTF-8"
        },
        body: JSON.stringify(user),
        mode: 'cors'
    };
    fetch('https://repo.re-ports.ru/app_test/Auth',options)
        .then(response => response.json())
        .then(data => expect(data).toMatchObject({
            success:true
        }))
        .catch(err => console.log(err))
});

test('get user data', () => {
   const options = {
       method: 'GET',
       credentials: 'include',
       mode: 'cors'
   };
    fetch('https://repo.re-ports.ru/app_test/UserData',options)
        .then(response => response.json())
        .then(data => console.log(data))
        .catch(err => console.log(err))
});