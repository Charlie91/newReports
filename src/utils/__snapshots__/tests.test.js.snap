// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="message text-white bg-danger text-center card"
>
  <div
    className="card-body card-block"
  >
    Текст ошибки
  </div>
</div>
`;
